packages/core/src/app/payment/paymentMethod/StripeUPEPaymentMethod.tsx

```diff
@@ -97,6 +98,14 @@ const StripeUPEPaymentMethod: FunctionComponent<
         );
     };
 
+    const shouldSavingCardsBeEnabled = (): boolean => {
```

Comment:
We have all these details in HostedWidgetPaymentMethod component, can we move this logic there?

Reply:
The idea was to make it more reusable, and give the possibility to pass shouldSavingCardsBeEnabled prop from payment method (if HostedWidgetPaymentMethod is used). For StripeUPE the conditions are described in shouldSavingCardsBeEnabled method and this way with passing shouldSavingCardsBeEnabled props gives other payment methods possibility to specify their own conditions.